src/platform/Darwin/DeviceInstanceInfoProviderImpl.cpp

```diff
@@ -23,15 +23,88 @@
 namespace chip {
 namespace DeviceLayer {
 
+CHIP_ERROR DeviceInstanceInfoProviderImpl::GetVendorName(char * buf, size_t bufSize)
+{
+    CHIP_ERROR err;
+    size_t vendorNameLen = 0; // without counting null-terminator
+    err = Internal::PosixConfig::ReadConfigValueStr(Internal::PosixConfig::kConfigKey_VendorName, buf, bufSize, vendorNameLen);
+    if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
+    {
+        VerifyOrReturnError(bufSize >= sizeof(CHIP_DEVICE_CONFIG_DEVICE_VENDOR_NAME), CHIP_ERROR_BUFFER_TOO_SMALL);
+        strcpy(buf, CHIP_DEVICE_CONFIG_DEVICE_VENDOR_NAME);
+    }
+
+    return err;
```

Comment:
The fallback path here seems broken, no?  Same for all the other methods in this file.